tests/integration_tests/test_wandb.py

```diff
@@ -28,114 +30,267 @@ def _multiobjective_func(trial: optuna.trial.Trial) -> Tuple[float, float]:
+@pytest.mark.parametrize("as_multirun", [True, False])
+def test_multiobjective_attributes_set_on_epoch(wandb: mock.MagicMock, as_multirun: bool) -> None:
 
-    wandb.config.update = mock.MagicMock()
+    wandb.sdk.wandb_run.Run = mock.MagicMock
+    trial_params = {"x": 1.1, "y": 2.2}
+    expected = {"direction": ["MINIMIZE", "MAXIMIZE"]}
+    expected_with_params = {"direction": ["MINIMIZE", "MAXIMIZE"], "x": 1.1, "y": 2.2}
 
-    wandbc = WeightsAndBiasesCallback()
     study = optuna.create_study(directions=["minimize", "maximize"])
+    wandbc = WeightsAndBiasesCallback(as_multirun=as_multirun)
+
+    study.enqueue_trial(trial_params)
     study.optimize(_multiobjective_func, n_trials=1, callbacks=[wandbc])
 
-    expected = {"direction": ["MINIMIZE", "MAXIMIZE"]}
-    wandb.config.update.assert_called_once_with(expected)
+    if as_multirun:
+        wandb.run = None
+        study.enqueue_trial(trial_params)
+        study.optimize(_multiobjective_func, n_trials=1, callbacks=[wandbc])
+        wandb.init().config.update.assert_called_once_with(expected_with_params)
+    else:
+        wandb.run.config.update.assert_called_once_with(expected)
 
 
 @mock.patch("optuna.integration.wandb.wandb")
-def test_log_api_call_count(wandb: mock.Mock) -> None:
+def test_log_api_call_count(wandb: mock.MagicMock) -> None:
 
-    wandb.log = mock.MagicMock()
+    wandb.sdk.wandb_run.Run = mock.MagicMock
 
+    study = optuna.create_study()
     wandbc = WeightsAndBiasesCallback()
     target_n_trials = 10
-    study = optuna.create_study()
     study.optimize(_objective_func, n_trials=target_n_trials, callbacks=[wandbc])
-    assert wandb.log.call_count == target_n_trials
+    assert wandb.run.log.call_count == target_n_trials
+
+    wandbc = WeightsAndBiasesCallback(as_multirun=True)
+    wandb.run.reset_mock()
+    _wrapped_logging_func = wandbc.track_in_wandb(
+        functools.partial(logging_objective_func, log_func=wandb.run.log)
+    )
+
+    study.optimize(_wrapped_logging_func, n_trials=target_n_trials, callbacks=[wandbc])
+
+    assert wandb.run.log.call_count == 2 * target_n_trials
+
+    wandb.run = None
+    study.optimize(_objective_func, n_trials=target_n_trials, callbacks=[wandbc])
+    assert wandb.init().log.call_count == target_n_trials
 
 
 @pytest.mark.parametrize(
-    "metric,expected", [("value", ["x", "y", "value"]), ("foo", ["x", "y", "foo"])]
+    "metric,as_multirun,expected",
+    [("value", False, ["x", "y", "value"]), ("foo", True, ["x", "y", "foo", "trial_number"])],
 )
 @mock.patch("optuna.integration.wandb.wandb")
-def test_values_registered_on_epoch(wandb: mock.Mock, metric: str, expected: List[str]) -> None:
+def test_values_registered_on_epoch(
+    wandb: mock.MagicMock, metric: str, as_multirun: bool, expected: List[str]
+) -> None:
+    def assert_call_args(log_func: mock.MagicMock, regular: bool) -> None:
+        kall = log_func.call_args
```

Comment:
Could you rename `kall` with a clearer name for Optuna's other maintainers? I could not understand the meaning of this variable.